### DISCUSSION TITLE
New Features Ideas

### DISCUSSION THREAD

**Discussion by ad-ha** [ad-ha/mg-saic-ha#59] [Ideas]
Hi everyone,

So based on my use case and in a way to improve the integration, I am thinking into add more functionalities and options.

Here are some of the ideas that pop into my mind.

## New auto data update intervals for buttons, switches and services.
If we use the front defrost switch, the data is currently updated 15 seconds after the action to make sure that it was successfully activated on the vehicle. Although, after 10 mints the car will turn off the front defrost automatically. 

I am thinking of including an option to get data again after 15 minutes to check that it has been deactivated and get the actual status (not 2 hours after, for example).
  
_Priority: 1
Difficulty: 1_ 

## Vehicle Options on Config Flow
Since MG is not to good at providing info on the vehicle data, I am thinking into storing the basic vehicle info and being able to get that data changed. Like, your car has sunroof but the API says it has not? You can activate it (or viceversa). The same for battery heating, battery capacity and so on...

This would allow to change the sensors available for each vehicle.

_Priority: 3
Difficulty: 5_

-----

Share any ideas that you might have and I may include those on the roadmap as well.

**Comment by brettmc** [comment]
As a UI enhancement, it would be nice to be able to know when an update is in progress (so my wife knows to be patient and not keep pressing the "update" button).
Perhaps something like a binary sensor that becomes true when an update is scheduled/enqueued/commenced, and off again when the request(s) complete. I could then use that sensor as a visibility condition on the update button.

**Reply by ad-ha** [reply]
Hi @brettmc

Thabks for the idea. Right now, you already have a next update sensor, that will tell you when is the next update due (in minutes).

Would that work for you?

**Reply by brettmc** [reply]
Sort of. What I observe is this:
* I have a button based on the "last update time" entity.
* It has an interaction to perform action "update vehicle data" when pressed.
* When I press the button, after some seconds (usually 10-15), the update occurs, and the last/next update times change.
* Until that happens, I don't _think_ there's any visual way to see if anything is "in progress" until the update occurs

This is super-low priority though, as I can just wait and see :)

**Reply by ad-ha** [reply]
I don't know if we are talking about the same thing, but on the last version you should have this:

![image](https://github.com/user-attachments/assets/5c5fad91-9138-423f-93e7-858dcaa647eb)
![image](https://github.com/user-attachments/assets/28ae2a85-07cb-40d3-8a38-48ce233ecb6c)

In my Lovelace I have this setup:
![image](https://github.com/user-attachments/assets/6f2d05f0-0b0b-4a96-8886-ed679b8369bc)

Whenever there is any change, either the 15 seconds updates or action updates (from 2 to 5 minutes) the next update sensor will display the expected time until the next update is triggered.

Was this what you were meaning?

**Reply by brettmc** [reply]
It's not quite what I was meaning, but that's exactly what I have in my lovelace too.

For those 15 second updates (update action triggered, a/c blowing or charging turned on/off, etc), I would like a way to know that "I got your request, it's in progress". It's purely a cosmetic/UI thing.
So for example, if there was a binary_sensor for update_in_progress, I could conditionally display an element that says `updating...`. It's an approximation of the iSmart app where doing something (here I've just turned on a/c blowing) has immediate visual feedback:
![Screenshot_20250107-104428](https://github.com/user-attachments/assets/6600e434-cbcc-4d29-9037-1a2eca55ee40)

**Reply by ad-ha** [reply]
Understood now. Thanks for the clarification.

I will see if I can add something like that.

**Comment by tuffgong31** [comment]
Hey,

I have an MG4.
support for heated steering wheel control on / off?

**Reply by ad-ha** [reply]
Hi @tuffgong31

By the moment, that option is not available on the API. It is possible to check the steering wheel heating level from the car response, but there is no service to control it.

If it comes up on the API, I will make it available as well.

**Reply by ad-ha** [reply]
BTW, are you able to control the steering wheel heating from the iSmart app?

**Reply by tuffgong31** [reply]
Hello,

yes I can.
Another question please
How often does your integration refresh the data? Is it different depending on the state of the car (off, on...)? 
And is it possible to force a refresh manually? Thanks

**Reply by ad-ha** [reply]
> How often does your integration refresh the data? Is it different depending on the state of the car (off, on...)?

It depends on the vehicle condition and the info received at each update. On the configuration menu of the integration you may see all different intervals, but as a summary:

![image](https://github.com/user-attachments/assets/d3f88ec7-fb02-4055-877a-d699b6f3314c)

Yes, you may trigger a manual update, either by service/action in developer tools or by pressing on this button inside the device data:

![image](https://github.com/user-attachments/assets/c0b40437-998c-4b4e-8985-6732eae860e0)

You may also add the button to your Lovelace.

**Reply by tbgoose** [reply]
> You may also add the button to your Lovelace.

Do you know if there are limits for hitting the API, or any risk increasing _frequency_ some of the charging related sensor update times to ~1 minute (or maybe even less)?

**Reply by townsmcp** [reply]
> Hey,
> 
> I have an MG4. support for heated steering wheel control on / off?

Just to add, this is now in the latest beta. But you will need to enable the option

**Reply by mstobie** [reply]
I have a 1 month old IM6 - divine. Since there is battery heating in the integration, how much of a leap is there for battery temperature sensor.  Not sure where it would take us but in Queensland I'm thinking about the effect of temperatures over 40C and deterioration on who knows what. In a previous career I witnessed plastics in low voltage connections inside computers melting slowly before succumbing suddenly at a critical time.

**Reply by townsmcp** [reply]
> I have a 1 month old IM6 - divine. Since there is battery heating in the integration, how much of a leap is there for battery temperature sensor. Not sure where it would take us but in Queensland I'm thinking about the effect of temperatures over 40C and deterioration on who knows what. In a previous career I witnessed plastics in low voltage connections inside computers melting slowly before succumbing suddenly at a critical time.

I am afraid that information isn’t in the API so we can’t map it

**Comment by mstobie** [comment]
Thanks James
Maybe on someone's wishlist.  It suggests you cannot overcook the battery
